packages/app/ui/recently-added-list.tsx

```diff
@@ -0,0 +1,110 @@
+import { FlatList, Image } from 'react-native'
+import { Pressable, Text, View } from 'app/design-system'
+import { useEffect, useState } from 'react'
+import { useQuery } from '@apollo/client'
+import { RECENTLY_ADDED } from 'app/api/beats'
+import Dollar from 'app/ui/icons/dollar'
+import Icon from '@expo/vector-icons/MaterialCommunityIcons'
+import { Entry, EntryPayload } from 'app/models'
+import { tw } from 'app/design-system/tailwind'
+
+type BeatEntry = {
+  imageUrl: string
+  videoUrl: string
+  description: string
+  title: string
+  id: string
+  artist: string
+  code: string
+  issuer: string
+}
+
+export default function RecentlyAddedList() {
+  const [nextPage, setNextPage] = useState(0)
+  const [data, setData] = useState<BeatEntry[]>([])
```

Comment:
I guess we would have a similar use case with pagination in other places as well. We could create a custom hook for that. I can help you with it, or I can add it afterward.